scan.hcl

```
@@ -0,0 +1,26 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+# Configuration for security scanner.
+# Run on PRs and pushes to `main` and `release/**` branches.
+# See .github/workflows/security-scan.yml for CI config.
+
+# To run manually, install scanner and then run `scan repository .`
+
+# Scan results are triaged via the GitHub Security tab for this repo.
+# See `security-scanner` docs for more information on how to add `triage` config
+# for specific results or to exclude paths.
+
+# .release/security-scan.hcl controls scanner config for release artifacts, which
+# unlike the scans configured here, will block releases in CRT.
+
+repository {
+  go_modules   = true
+  osv          = true
+  oss_index    = true
+  nvd          = true
```

My goal here was mainly parity w/ the `binary` scan.

Security: is this a reasonable set? Are there any gaps between `repository` and `binary` for this sort of config, or is `repository` strictly a superset, even in our case where we have internal dependencies on submodules within this repo?

---

I suggest we use the following:

```suggestion
repository {
  go_modules   = true
  npm          = true
  osv          = true
```

I think `osv` will provide us with the coverage we want here for the Go and NPM (Yarn, etc) ecosystems. We can also simply remove `oss_index` and `nvd` here, instead of setting them to `false`. 

We can't enable `npm` for the binary targets today, because we don't yet embed the relevant file (`yarn.lock`) in the Consul binary. That is something we may investigate in the new year.